src/os/bluestore/BlueStore.cc

```diff
@@ -291,31 +283,10 @@ static void get_object_key(const ghobject_t& oid, string *key)
   _key_encode_shard(oid.shard_id, key);
   _key_encode_u64(oid.hobj.pool + 0x8000000000000000ull, key);
   _key_encode_u32(oid.hobj.get_bitwise_key_u32(), key);
-  key->append(".");
 
   append_escaped(oid.hobj.nspace, key);
-
-  if (oid.hobj.get_key().length()) {
-    // is a key... could be < = or >.
-    // (ASCII chars < = and > sort in that order, yay)
-    if (oid.hobj.get_key() < oid.hobj.oid.name) {
-      key->append("<");
-      append_escaped(oid.hobj.get_key(), key);
-      append_escaped(oid.hobj.oid.name, key);
-    } else if (oid.hobj.get_key() > oid.hobj.oid.name) {
-      key->append(">");
-      append_escaped(oid.hobj.get_key(), key);
-      append_escaped(oid.hobj.oid.name, key);
-    } else {
-      // same as no key
-      key->append("=");
-      append_escaped(oid.hobj.oid.name, key);
-    }
-  } else {
-    // no key
-    key->append("=");
-    append_escaped(oid.hobj.oid.name, key);
-  }
+  append_escaped(oid.hobj.get_key(), key);
+  append_escaped(oid.hobj.oid.name, key);
```

Comment:
This block is here for a reason: to avoid encoding the name twice when key == name (i.e., almost all of the time).  For this we need the < > = separators.